src/Core/Framework/Gateway/Context/Command/Handler/AbstractContextGatewayCommandHandler.php

```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Gateway\Context\Command\Handler;
+
+use Shopware\Core\Framework\Gateway\Context\Command\AbstractContextGatewayCommand;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+/**
+ * @template TCommand of AbstractContextGatewayCommand
+ */
+#[Package('framework')]
+abstract class AbstractContextGatewayCommandHandler
+{
+    /**
+     * @param TCommand $command
+     * @param array<string, mixed> $parameters
+     */
+    abstract public function handle(AbstractContextGatewayCommand $command, SalesChannelContext $context, array &$parameters): void;
+
+    /**
+     * @return array<class-string<AbstractContextGatewayCommand>>
```

Comment:
So supportedCommands and type have to be equal:
```suggestion
     * @return array<class-string<TCommand>>
```